Adding Connecticut County Equivalents

brendensm opened the pull request:
This pull request closes issue #14 asking for added support for Connecticut's new planning regions. This PR:

- adds a new crosswalk specific to Connecticut's ZCTAs
- changes the logic of get_zcta_by_county to support the Connecticut-specific planning region codes
- adds a script to the inst folder to create this crosswalk from tigris shape files
- updates documentation to reflect this change

Feel free to make any changes as necessary!

arilamstein commented:
I left MarketBridge shortly before this PR was made, and so did not get a notification about this PR (or the issue that caused it to get created). I assume that the notification went to my market-bridge email, which I no longer have access to.

I have not worked in R since leaving MarketBridge. So I think that the best way to handle this PR might be to combine it with having someone take over as maintainer of the package. Are you interested in filling that role?

arilamstein commented on DESCRIPTION:
```diff
@@ -19,7 +19,8 @@ Imports:
 Depends:
     R (>= 3.5.0)
-RoxygenNote: 7.2.3
+RoxygenNote: 7.3.1
```
Please also update the package version, and also add yourself as a contributor. This page discusses the various "roles" that CRAN likes to have defined, and what they actually mean: https://r-pkgs.org/description.html#sec-description-authors-at-r

Note that if you wind up taking on the role of package maintainer, I believe that you will need to list yourself as having the role of "creator" (even though I created the package).

brendensm replied:
Hi Ari, thanks for your thoughtful comments on this pull request. I would be interested in taking over the role of package maintainer, if that is ok with you. I will continue to address your comments below and commit the changes accordingly.

arilamstein replied:
> Hi Ari, thanks for your thoughtful comments on this pull request. I would be interested in taking over the role of package maintainer, if that is ok with you. I will continue to address your comments below and commit the changes accordingly.

I am very interested in that - thank you!

arilamstein commented on R/zcta.R:
```diff
@@ -66,15 +66,38 @@ get_zctas_by_county = function(counties) {
+
+  }else{
+
+    message(paste("Using column", col))
```
If, up above, we're specifying the data structure we're pulling from, then we should probably be doing that here too.

More broadly, I think that this PR kind of "sneaks in" some new functionality, and I would rather we be more explicit about it. Ways we can achieve that are:

1. Having a sentence or two about this new data structure, and the problem it solves, in the README. And perhaps even expanding the `Description:` field of the DESCRIPTION file to mention it.
2. Doing the same in at least one of the vignettes. For example, perhaps the first vignette should mention the issue that caused this PR to be needed. Vignette 3 was designed for whoever winds up maintaining the package when the next ZCTAs are published. Perhaps it's worth explicitly saying in that vignette that this functionality probably doesn't need to be updated for the 2030 data, and that the data structure you're adding in can safely be deleted then. (Assuming that my understanding is correct, and that it can be deleted then).

arilamstein commented on R/data.R:
```diff
@@ -18,4 +18,18 @@ NULL
+
+#' Crosswalk between Connecticut County Equivalents (Planning Regions) and ZCTAs
+#'
+#' The complete dataset in ?ct_cw contains information Connecticut's new planning
```
This is good, but I think it's worth being more explicit about what problem this data structure solves. For example, here is the link linked to in the issue:

https://www.census.gov/programs-surveys/geography/technical-documentation/county-changes.2020.html#list-tab-957819518

And here is the first link on that page:
https://www.federalregister.gov/documents/2022/06/06/2022-12063/change-to-county-equivalents-in-the-state-of-connecticut

It might be worth including links to one or both of these sources, so people have some context. I think it's probably also worth mentioning the date at which the Planning Regions replaced the Counties (it looks like January 2022?)

brendensm committed:
Fix CRAN check errors to prevent archival

  The tidycensus vignette was failing CRAN checks on 9 platforms because it
  called get_acs() at build time, which now requires a Census API key that
  CRAN's build machines do not have. This addresses that error along with the
  accompanying NOTE and data-compression WARNING.

  - Pre-compute vignettes/a02_workflow-tidycensus.Rmd using the .Rmd.orig
    pattern: the live source (with get_acs() calls) lives in the .Rmd.orig
    file, and the built .Rmd ships pre-rendered output. CRAN now builds the
    vignette with no API key or network access, while readers still see the
    full results. Add vignettes/precompile.R to regenerate it locally.
  - Declare Depends: R (>= 4.1.0), since the package source uses the native
    pipe |> (was incorrectly declared as R >= 3.5.0).
  - Add LazyDataCompression: xz and resave data/*.rda with xz to resolve the
    "significantly better compression" WARNING.
  - Add .Rbuildignore for the .orig source, precompile.R, and RStudio files.
  - Bump version to 2.0.1.

brendensm commented:
@arilamstein This should close the loop on the feedback you provided here. Open to other changes as you see fit.

brendensm committed:
Update maintainer, email, and URL per Ari's review

brendensm committed:
Add cran-comments.md

brendensm committed:
fix redirecting marketbridge url
